### ISSUE TITLE
Clean up Pattern Library components and styling

### ISSUE THREAD

**Issue by lyzadanger** [hypothesis/frontend-shared#102]
The Pattern Library components and styles have grown organically during development and have served their purpose well enough but need some pruning and cleanup.

* The `PatternExample*` components are a little verbose and redundant in their usage
* The tabular layout in `PatternExamples` isn't quite right, has issues with column width/consistency, and isn't responsive
* More granular control over what bits of source get JSX-to-string-ed would be nice
* The styles should stop using element selectors; the stylesheet needs cleanup overall

Basically the goals are:

* Make authorship using these components as convenient as possible, and as reasonably terse as possible
* Make the styling follow app conventions
* Make sure things are responsive and as attractive as reasonably possible
* Don't break existing application pattern-library pages
